server/py/services/alerts/crud/alerts.py

```diff
@@ -46,86 +47,46 @@ def store_alert(
         )
 
         self._validate_alert(alert_data, name, project)
-
-        if alert_data.criteria is None:
-            alert_data.criteria = mlrun.common.schemas.alert.AlertCriteria()
+        alert_data.criteria = (
+            alert_data.criteria or mlrun.common.schemas.alert.AlertCriteria()
+        )
 
         if existing_alert is not None:
-            self._delete_notifications(existing_alert)
-            self._get_alert_by_id_cached().cache_remove(session, existing_alert.id)
-
-            for kind in existing_alert.trigger.events:
-                services.alerts.crud.Events().remove_event_configuration(
-                    project, kind, existing_alert.id, existing_alert.entities.ids[0]
-                )
-
-            # preserve the original creation time and id of the alert so that modifying the alert does not change them
-            alert_data.created = existing_alert.created
-            alert_data.id = existing_alert.id
-
-            # set the updated field to reflect the latest modification time of the alert
-            alert_data.updated = mlrun.utils.now_date()
-
-            # Enrich the old alert with existing state
-            existing_alert.state = mlrun.common.schemas.AlertActiveState.INACTIVE
-            if existing_alert_state and existing_alert_state.to_dict()["active"]:
-                existing_alert.state = mlrun.common.schemas.AlertActiveState.ACTIVE
-        else:
-            num_alerts = (
-                framework.utils.singletons.db.get_db().get_num_configured_alerts(
-                    session
-                )
+            self._handle_existing_alert(
+                session, project, existing_alert, alert_data, existing_alert_state
             )
-            if num_alerts >= mlconfig.alerts.max_allowed:
-                raise mlrun.errors.MLRunPreconditionFailedError(
-                    f"Allowed number of alerts exceeded: {num_alerts}"
-                )
+        else:
+            self._check_alerts_limit(session)
 
         self._validate_and_mask_notifications(alert_data)
-
         new_alert = (
             framework.utils.singletons.db.get_db().store_alert(session, alert_data)
             if existing_alert
             else framework.utils.singletons.db.get_db().create_alert(
                 session, alert_data
             )
         )
-
-        for event_kind in new_alert.trigger.events:
-            services.alerts.crud.Events().add_event_configuration(
-                project, event_kind, new_alert.id, new_alert.entities.ids[0]
-            )
+        self._add_event_configurations(project, new_alert)
 
         # if the alert already exists we should check if it should be reset or not
         if existing_alert is not None:
-            should_reset, reset_reason = self._should_reset_alert(
-                existing_alert, alert_data, force_reset
+            self._reset_if_needed(
```

Comment:
I think it's better to pass args as named here. It improves readability and eliminates relying on the order. 

Same for `_handle_existing_alert`